### PR TITLE
feature(factory): syncCollection/CollectionGroup in factory service

### PR DESCRIPTION
**Pull request by loicmarie**

### CODE REVIEW

File: projects/akita-ng-fire/src/lib/factory/factory.service.ts [hunk 1]
```diff
@@ -125,7 +184,7 @@ export class FactoryService<S extends EntityState<any, string>> {
   ): Promise<void>;
   async update(
     idOrEntity: string | Partial<getEntityType<S>>,
-    newStateOrOption?: Partial<getEntityType<S>> | WriteOptions,
+    newStateOrOption?: Partial<getEntityType<S>>,
```
Comment (GrandSchtroumpf) [inline projects/akita-ng-fire/src/lib/factory/factory.service.ts]: I don't understand why you remove `WriteOptions` here. You should be able to do :
```
service.update({ id: '1', title: 'new title' }, { write: batch });
```

File: projects/akita-ng-fire/src/lib/utils/sync-from-action.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ export function syncFromAction<S>(
 
 
 export function syncStoreFromAction<S>(
+  idKey: string,
```
Comment (GrandSchtroumpf) [inline projects/akita-ng-fire/src/lib/utils/sync-from-action.ts]: Should be last optional argument : 
```typescript
export function syncStoreFromAction<S>(
  storeName: string,
  actions: DocumentChangeAction<getEntityType<S>>[],
  idKey: string = 'id'
) { ... }
```

File: projects/akita-ng-fire/src/lib/factory/factory.service.ts [hunk 3]
```diff
@@ -30,17 +51,55 @@ export class FactoryService<S extends EntityState<any, string>> {
...
   }
 
-  syncStore(name: string, queryFn?: QueryFn) {
-    return this.db
-      .collection<getEntityType<S>>(this.path, queryFn)
-      .stateChanges().pipe(
-        withTransaction(actions => syncStoreFromAction(name, actions))
-      );
+  /** Stay in sync with the collection or a fractio of it */
+  syncCollection(name: string, path?: string | Observable<string> | QueryFn): Observable<DocumentChangeAction<getEntityType<S>>[]>;
```
Comment (GrandSchtroumpf) [inline projects/akita-ng-fire/src/lib/factory/factory.service.ts]: Change "name" to "storeName" for better understanding.

Comment (loicmarie) [reply projects/akita-ng-fire/src/lib/factory/factory.service.ts]: 6949b37

File: projects/akita-ng-fire/src/lib/utils/sync-from-action.ts (hunk 2 above)
Comment (loicmarie) [reply projects/akita-ng-fire/src/lib/utils/sync-from-action.ts]: 6949b37

File: projects/akita-ng-fire/src/lib/factory/factory.service.ts (hunk 1 above)
Comment (loicmarie) [reply projects/akita-ng-fire/src/lib/factory/factory.service.ts]: 6949b37

**Pull request merged**